helm/templates/daemonset.yaml

```diff
@@ -28,36 +30,40 @@ spec:
             - {{ $k }}.efs.{{ $v.region }}.amazonaws.com
       {{- end }}
     {{- end }}
-    {{- if .Values.imagePullSecrets }}
+    {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
-      {{- range .Values.imagePullSecrets }}
-        - name: {{ . }}
-      {{- end }}
+        {{- toYaml . | nindent 8 }}
     {{- end }}
+      serviceAccountName: {{ template "aws-efs-csi-driver.serviceAccountName" . }}
```

Comment:
There isn't anything technically wrong with using template here; however, template can't be used in pipelines so I tend to use include instead.